lib/authentication/service/auth_provider.dart

```diff
@@ -202,6 +204,24 @@ class AuthProvider with ChangeNotifier {
     }).then((_) => true);
   }
 
+  Future<bool> changePassword({String password, BuildContext context}) async {
+    bool result = false;
+    await _firebaseUser.updatePassword(password).then((_) {
+      AppToast.show(S.of(context).messageChangePasswordSuccess);
+      result = true;
+    });
```

Comment:
From my tests this works. The dialog that calls this method closes only if the result is true.

Reply:
Ahh, you called `await` *and* `then`. I agree that it works, but it's kind of pointless.
Also, this will actually *always* return `true` because you don't check whether `changePassword` completes successfully. You need to use `catchError` or a try-catch block and get rid of the `then`.

Also, for consistency, the success message should be displayed from the UI.